### ISSUE TITLE
[feature request] audio delay

### ISSUE THREAD

**Issue by zaza42** [mifi/lossless-cut#216]
Please add an option for setting audio delay, a.k.a. changing a/v sync!

**Comment by mifi** [comment]
Then we need to `-ss` differently for audio vs video. Not sure if this is easy. Maybe something like this:
https://stackoverflow.com/questions/55147398/how-to-specify-an-offset-and-a-seek-position-for-a-video-stream-of-a-mp4-contain

**Comment by zaza42** [comment]
@mifi : Yes, this is what i exactly use.

**Comment by chasylexus** [comment]
Searched exactly the same.
My use case:
I have screencast without audio and audio corresponded to screencast. Audio begins with several seconds of silence and should be synced with video before merging. It will be super-nice to have this feature.
Sometimes audio also have a little bit high speed that video (several seconds after several hours) and should be `scaled` but this is not so common case I think.

**Comment by mifi** [comment]
https://wjwoodrow.wordpress.com/2013/02/04/correcting-for-audiovideo-sync-issues-with-the-ffmpeg-programs-itsoffset-switch/

**Comment by Binger-cn** [comment]
I need this option very much !
I often shot badminton matches, each time result in mp4 files usually 40GB+ per file. However, I used bluetooth audio devices, so that audio is 200ms later than video. And I need to cut each file into 200+ segments for later use.  
I used LosslessCut to cut the original file, but in each segment the audio is still delayed 200ms.   The solution I'm now using is that first I use mkvtoolnix to create a temp file, also 40GB+, and cut the mkv file in LosslessCut.  This procedure really costs too much time and disk space.
So I wish LosslessCut could provide an option to delay an audio track some miliseconds, with a positive or negative number.

**Comment by mifi** [comment]
FYI @gyarmish posted a manual workaround for adjusting individual tracks, see https://github.com/mifi/lossless-cut/issues/929#issue-1053948979

**Comment by gyarmish** [comment]
Thanks for that mention mifi. 

In addition to its usefulness - part of the reason for the post was to show how it can be implemented in a new release - its quite tedious manually but it can be automate this behind the scenes using the standard lossless functions. 

Imagine having to do that for multiple short videos. It will get done but but if those steps can be programmed in it would be very useful.

**Comment by mifi** [comment]
Yes if I cannot find a way for ffmpeg to shift the offsets, then extracting, cutting and re-importing individual tracks like you outlined could be a workaround however I would imagine that it's much slower (more CPU/resources) than letting ffmpeg do it internally

**Comment by Typhoon859** [comment]
Has this feature been added?  I can see that the program is well written and I appreciate all the work that's been put into it.  This ability to shift/synchronize the audio is a must however.  On AVIDemux, there's a simple option for the audio to enter a ms delay amount (+ or -) and it's always worked great.  Any word on how to achieve this simply and easily using LosslessCut?  Thanks in advance!

**Comment by mifi** [comment]
Not yet! But if AVIDemux can do it, maybe that's a workaround for now?

**Comment by Typhoon859** [comment]
> Not yet! But if AVIDemux can do it, maybe that's a workaround for now?

Well sure, but... LosslessCut :(

**Comment by mifi** [comment]
don't worry - this is one of the most popular feature requests, so i will definitely add it:)

**Comment by Typhoon859** [comment]
> don't worry - this is one of the most popular feature requests, so i will definitely add it:)

Awesome.  That's great to hear man; thank you.

**Comment by jauling** [comment]
Adding this functionality would also improve the workflow when adding music to a clip. Often the audio I want to add is only a snippet of an entire track. Being able to seek and find the right starting point would be killer, thanks!

**Comment by romanr** [comment]
Thank you @mifi for the wonderful app. Been using it for years and this is the first time the issue I'd like to vote for!

**Comment by mifi** [comment]
In #2528 we've been testing `-itsoffset 0` on files that have a start_time offset between the audio and video track (e.g. audio delay), and for some files it removes (most of the) time offset. I don't quite understand it, because `-itsoffset 0` is not supposed to do anything (`-itsoffset` flag is supposed to add the specified differential offset to the existing offset, according to ffmpeg docs).

**Comment by mifi** [comment]
i've now added a basic `-tsoffset` option per file in the Tracks panel (edit file metadata). can be tested in the next nightly build!
